src/app/frontend/replicasetdetail/replicasetdetail_state.js

```diff
@@ -57,16 +58,23 @@ export default function stateConfig($stateProvider) {
 }
 
 /**
+ *
  * @param {!StateParams} $stateParams
  * @param {!angular.$resource} $resource
- * @return {!angular.$q.Promise}
+ * @return {!angular.Resource<!backendApi.ReplicaSetDetail>}
  * @ngInject
  */
-export function resolveReplicaSetDetails($stateParams, $resource) {
-  /** @type {!angular.Resource<!backendApi.ReplicaSetDetail>} */
-  let resource = $resource('/api/replicasets/:namespace/:replicaSet', $stateParams);
+export function getReplicaSetDetailsResource($stateParams, $resource) {
+  return $resource('/api/replicasets/:namespace/:replicaSet', $stateParams);
+}
 
-  return resource.get().$promise;
+/**
+ * @param {!angular.Resource<!backendApi.ReplicaSetDetail>} replicaSetDetailResource
+ * @return {!angular.$q.Promise}
+ * @ngInject
+ */
+export function resolveReplicaSetDetails(replicaSetDetailResource) {
+  return replicaSetDetailResource.get().$promise;
```

Comment:
@bryk: What do you think about this solution to keep resource endpoint initialization in one place?